codec-haproxy/src/main/java/io/netty/handler/codec/haproxy/HAProxyMessage.java

```diff
@@ -306,17 +286,18 @@ private static HAProxyTLV readNextTLV(final ByteBuf header) {
      * @return                           {@link HAProxyMessage} instance
      * @throws HAProxyProtocolException  if any portion of the header is invalid
      */
-    static HAProxyMessage decodeHeader(String header) {
-        if (header == null) {
+    static HAProxyMessage decodeStringHeader(ByteBuf header) {
+        String headerStr = header.toString(CharsetUtil.US_ASCII);
+        if (headerStr == null) {
```

Comment:
`toString(...)` will never return null so this does not look correct. Maybe it should check if the buffer is empty ?

Reply:
Done